torch/csrc/dynamo/eval_frame.c

```diff
@@ -320,6 +320,39 @@ inline static CacheEntry* get_cache_entry(PyCodeObject* code) {
   return extra;
 }
 
+PyObject* _debug_get_cache_entry_list(PyObject* self, PyObject* args) {
+  PyObject* object;
+  if (!PyArg_ParseTuple(args, "O", &object)) {
+    return NULL;
+  }
+  if (!PyCode_Check(object)) {
+    PyErr_SetString(PyExc_TypeError, "expected a code object!");
+  }
+  PyCodeObject* code = (PyCodeObject*)object;
+
+  CacheEntry* current_node = get_cache_entry(code);
+
+  PyObject* outer_list = PyList_New(0);
+  if (!outer_list) {
+    return NULL;  // Return NULL if failed to create list
+  }
+  while (current_node != NULL) {
```

Comment:
I believe your crash is coming from this being missing:
```suggestion
  while (current_node != NULL && current_node != SKIP_CODE) {
```

Reply:
Fixed in [89ccb54](https://github.com/pytorch/pytorch/pull/106673/commits/89ccb542e3d187a7c77dcc275cbd48a8faf77bf4). Huge thanks!